packages/graphile-utils/src/makeExtendSchemaPlugin.ts

```diff
@@ -480,14 +481,24 @@ export default function makeExtendSchemaPlugin(
     builder.hook("GraphQLSchema", (schema, build, _context) => {
       const {
         [`ExtendSchemaPlugin_${uniqueId}_typeExtensions`]: typeExtensions,
+        [`ExtendSchemaPlugin_${uniqueId}_newTypes`]: newTypeDefinitions,
       } = build;
+      const newTypes = newTypeDefinitions.map(
+        ({ definition }: { definition: TypeDefinitionNode }) =>
+          build.getTypeByName(definition.name.value)
+      );
+
       return {
         ...schema,
         directives: [
           ...(schema.directives || build.graphql.specifiedDirectives || []),
           ...typeExtensions.GraphQLSchema.directives,
         ],
-        types: [...(schema.types || []), ...typeExtensions.GraphQLSchema.types],
+        types: [
+          ...(schema.types || []),
+          ...typeExtensions.GraphQLSchema.types,
+          ...newTypes,
```

Comment:
This is great and very simple, but the problem is that it significantly changes the generated schema order - putting the types added by makeExtendSchemaPlugin above everything else.

I _think_ that `graphql` walks the types in order, so we may be able to solve this fairly simply by just ensuring that the Query, Mutation and Subscription types are processed first. That might look something like this:

```suggestion
          ...([
            build.getTypeByName(inflection.builtin("Query")),
            build.getTypeByName(inflection.builtin("Mutation")),
            build.getTypeByName(inflection.builtin("Subscription")),
          ].filter(_ => _)),
          ...(schema.types || []),
          ...typeExtensions.GraphQLSchema.types,
          ...newTypes,
```